src/Api/Tools/Models/Response/MemberAccessReportModel.cs

```diff
@@ -0,0 +1,150 @@
+    public string GroupName { get; set; }
+    public string CollectionName { get; set; }
+    public int ItemCount { get; set; }
+    public bool ReadOnly { get; set; }
+    public bool HidePasswords { get; set; }
+    public bool Manage { get; set; }
+
+    // internal to not expose 
+    internal Guid? UserGuid { get; set; }
+}
+
+/// <summary>
+/// Contains the collections and group collections a user has access to including
+/// the permission level for the collection and group collection. 
+/// </summary>
+public class MemberAccessReportResponseModel
+{
+    public string UserName { get; set; }
+    public string Email { get; set; }
+    public bool TwoFactorEnabled { get; set; }
+    public bool AccountRecoveryEnabled { get; set; }
+    public int GroupsCount { get; set; }
+    public int CollectionsCount { get; set; }
+    public int TotalItemCount { get; set; }
+    public IEnumerable<MemberAccessReportAccessDetails> AccessDetails { get; set; }
+
+    /// <summary>
+    /// Generates a report for all members of an organization. Containing summary information
+    /// such as item, collection, and group counts. As well as detailed information on the
+    /// user and group collections along with their permissions
+    /// </summary>
+    /// <param name="orgUsers">Organization user details collection</param>
+    /// <param name="orgGroups">Organization groups collection</param>
+    /// <param name="orgCollectionsWithAccess">Collections for the organization and the groups/users and permissions</param>
+    /// <param name="orgItems">Cipher items for the organization with the collections associated with them</param>
+    /// <param name="organizationUsersTwoFactorEnabled">Organization users two factor status</param>
+    /// <param name="orgAbility">Organization ability for account recovery status</param>
+    /// <returns>List of the MemberAccessReportResponseModel</returns>;
+    public static IEnumerable<MemberAccessReportResponseModel> CreateReport(
+        IEnumerable<OrganizationUserUserDetails> orgUsers,
+        ICollection<Group> orgGroups,
+        ICollection<Tuple<Collection, CollectionAccessDetails>> orgCollectionsWithAccess,
+        IEnumerable<CipherOrganizationDetailsWithCollections> orgItems,
+        IEnumerable<(OrganizationUserUserDetails user, bool twoFactorIsEnabled)> organizationUsersTwoFactorEnabled,
+        OrganizationAbility orgAbility)
+    {
+        // Create a dictionary to lookup the group names later. 
+        var groupNameDictionary = orgGroups.ToDictionary(x => x.Id, x => x.Name);
+
+        // Get collection counts into a dictionary
+        var groupCollectionItems = orgItems.SelectMany(x => x.CollectionIds, (x, b) => new { CipherId = x.Id, CollectionId = b }).GroupBy(y => y.CollectionId, (key, g) => new { CollectionId = key, ItemCount = g.Count() });
+        var collectionCountDictionary = groupCollectionItems.ToDictionary(x => x.CollectionId, x => x.ItemCount);
```

Comment:
I suggest just chaining this at the end of the preceding variable declaration. As long as each chained operator is on a new line, I think that's easier to follow than declaring additional variables.

Reply:
Sounds good. That's always a thing I struggle with deciding. Sometimes I think having a new variable can help make it readable and easier to debug. Sometimes just stringing together is better. I appreciate the input and will change.

Reply:
It's definitely a matter of opinion; certainly it makes debugging harder. Chaining is useful to me because it shows you don't care (and aren't going to use) the intermediate results, you just care about the outcome at the end of the chain. That's particularly useful here where we already have a lot of variables in scope.

You haven't made this change but it's also non-blocking so I'll leave it up to you.

Reply:
I see that you needed to keep the `collectionItems` to count them properly later. That makes sense to me!